src/api/controllers/auth.js

```diff
@@ -26,6 +28,35 @@ const registerController = async (req, res) => {
   return res.status(StatusCodes.CREATED).json({ message: 'User created' });
 };
 
+const forgotPasswordController = async (req, res) => {
+  const { email } = req.body;
+
+  if (!email)
+    return res
+      .status(StatusCodes.BAD_REQUEST)
+      .json({ message: 'Email can not be empty' });
+
+  const foundUser = await User.findOne({ email });
+
+  if (!foundUser) throw new NotFound('No user found with this email');
+
+  const token = foundUser.createPasswordResetToken();
+  foundUser.passwordResetToken = token;
+  const result = await foundUser.save();
+  const url = `http://localhost:${process.env.PORT}/api/v1/auth/reset-password/${foundUser.email}/${token}`;
+
+  const data = { url };
+  const text = ` Here is your forgot password link: ${url} 
+  If you did not make this request, kindly ignore. `;
+
+  sendMail('Forgot Password', email, text, 'forgot-password.hbs', data);
```

Comment:
the subject should be in this format 
Your password reset link only valid for 10 mins